File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Map<Integer,Integer> map = new HashMap<>();
        for (int num: nums){
            int count = map.getOrDefault(num,0);
            map.put(num,count+1);
        }
        for (int num: map.keySet()){
            int count = map.get(num);
            if (count>1) return true; 
        }
        return false; 
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # edge case
        if len(nums) == 1 : 
            return 1

        unique_arr = []
        for num in nums : 
            if num in unique_arr :
                continue 
            else : 
                unique_arr.append(num)
        
        for i,num in enumerate(unique_arr):
            nums[i] = unique_arr[i]
        
        return len(unique_arr)